tests/Aspire.Dashboard.Tests/Middleware/ValidateTokenMiddlewareTests.cs

```diff
@@ -0,0 +1,107 @@
+        var response = await host.GetTestClient().GetAsync("/login?t=test");
+        Assert.Equal("/", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task ValidateToken_NotBrowserTokenAuth_RedirectedToReturnUrl()
+    {
+        using var host = await SetUpHostAsync(FrontendAuthMode.Unsecured, string.Empty);
+        var response = await host.GetTestClient().GetAsync("/login?t=test&returnUrl=/test");
+        Assert.Equal("/test", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task ValidateToken_BrowserTokenAuth_WrongToken_RedirectsToLogin()
+    {
+        using var host = await SetUpHostAsync(FrontendAuthMode.BrowserToken, "token");
+        var response = await host.GetTestClient().GetAsync("/login?t=wrong");
+        Assert.Equal("/login", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task ValidateToken_BrowserTokenAuth_WrongToken_RedirectsToLogin_WithReturnUrl()
+    {
+        using var host = await SetUpHostAsync(FrontendAuthMode.BrowserToken, "token");
+        var response = await host.GetTestClient().GetAsync("/login?t=wrong&returnUrl=/test");
+        Assert.Equal("/login?returnUrl=%2ftest", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task ValidateToken_BrowserTokenAuth_RightToken_RedirectsToHome()
+    {
+        using var host = await SetUpHostAsync(FrontendAuthMode.BrowserToken, "token");
+        var response = await host.GetTestClient().GetAsync("/login?t=token");
+        Assert.Equal("/", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task ValidateToken_BrowserTokenAuth_RightToken_RedirectsToReturnUrl()
+    {
+        using var host = await SetUpHostAsync(FrontendAuthMode.BrowserToken, "token");
+        var response = await host.GetTestClient().GetAsync("/login?t=token&returnUrl=/test");
+        Assert.Equal("/test", response.Headers.Location?.OriginalString);
+    }
+
+    private static async Task<IHost> SetUpHostAsync(FrontendAuthMode authMode, string expectedToken)
+    {
+        return await new HostBuilder()
+            .ConfigureWebHost(webBuilder =>
+            {
+                webBuilder
+                    .UseTestServer()
+                    .ConfigureServices(services =>
+                    {
+                        services.AddRouting();
+                        services.AddAuthentication().AddCookie();
+                        services.AddSingleton(GetOptionsMonitorMock(authMode, expectedToken));
+                    })
+                    .Configure(app =>
+                    {
+                        app.UseMiddleware<ValidateTokenMiddleware>();
+                    });
+            })
+            .StartAsync();
+    }
+
+    private static IOptionsMonitor<DashboardOptions> GetOptionsMonitorMock(FrontendAuthMode authMode, string expectedToken)
```

Comment:
Change this to set options normally and remove NSubstitute dependency.

Should be able to do something like this:

```cs
services.AddOptions();
services.Configure<DashboardOptions>(o =>
{
    // configure dashboard options
});
```